src/v/cluster/partition_balancer_planner.cc

```diff
@@ -865,6 +867,42 @@ void partition_balancer_planner::get_rack_constraint_repair_actions(
     }
 }
 
+/**
+ * This is the place where we decide about the order in which partitions will be
+ * moved in the case when node disk is being full.
+ */
+size_t partition_balancer_planner::calculate_full_disk_partition_move_priority(
+  model::node_id node_id,
+  const reassignable_partition& p,
+  const request_context& ctx) {
+    static constexpr size_t min_priority = 0;
+    static constexpr size_t max_priority = 1000000;
+
+    if (p.size_bytes() < ctx.config().min_partition_size_threshold) {
+        return min_priority;
```

Comment:
Not sure if this is useful, but I guess we should still make some priority distinction between partitions under the threshold. I was thinking about something like `threshold/size` if `size < threshold` (need to wait this number of min partition move times to see a disk usage drop of `threshold` if moving small partitions 1-by-1) else `size/threshold` (need to wait this number of min partition move times before we see any disk usage drop). This is with min-based priority order.